src/org/labkey/test/tests/DomainDesignerTest.java

```diff
@@ -1498,6 +1507,172 @@ public void addUpdateRemoveConditionalFormat() throws Exception
+
+        String roundTripSampleType = "testSampleTypeWithImportedFields";
+        File exportFile = domainFormPanel.clickExportFields();
+        CreateSampleTypePage sampleTypeCreatePage = CreateSampleTypePage.beginAt(this, getProjectName());
+        sampleTypeCreatePage.setName(roundTripSampleType);
+        sampleTypeCreatePage.getFieldsPanel()
+                .setInferFieldFile(exportFile);
+        sampleTypeCreatePage.clickSave();
+
+        DomainDesignerPage sampleTypeDesignerPage = DomainDesignerPage.beginAt(this, getProjectName(), "exp.materials", roundTripSampleType);
+        File roundTripFile = sampleTypeDesignerPage.fieldsPanel().clickExportFields();
+        List<PropertyDescriptor> roundTrippedFields = getFieldsFromExportFile(roundTripFile);
+
+        for (PropertyDescriptor intendedField : dgen.getColumns())
+        {
+            String fieldName = intendedField.getName();
+            if (!fieldName.equals("name"))  // don't expect name col to export from sampletype, it is special
+            {
+                PropertyDescriptor exportedField = roundTrippedFields.stream().filter(a -> a.getName().equals(fieldName))
+                        .findFirst().orElse(null);
+                assertNotNull("expected field [" + intendedField.getName() + "] was not exported", exportedField);
+                verifyExpectedFieldProperties(intendedField, exportedField);
+            }
+        }
+    }
+
+    private void verifyExpectedFieldProperties(PropertyDescriptor intendedField,  PropertyDescriptor actualField)
+    {
+        log("verifying properties for field [" +intendedField.getName()+ "]");
+        assertThat(actualField.getDescription(), is(intendedField.getDescription()));
+        assertThat(actualField.getLabel(), is(intendedField.getLabel()));
+        assertThat(actualField.getRequired(), is(intendedField.getRequired()));
+        assertThat(actualField.getHidden(), is(intendedField.getHidden()));
+        assertThat(actualField.getMvEnabled(), is(intendedField.getMvEnabled()));
+        assertThat(actualField.getDimension(), is(intendedField.getDimension()));
+        if (intendedField.getPHI() != null)
+            assertThat(actualField.getPHI(), is(intendedField.getPHI()));
+        assertThat(actualField.getMeasure(), is(intendedField.getMeasure()));
+        assertThat(actualField.getFormat(), is(intendedField.getFormat()));
+        for (ConditionalFormat intendedFormat : intendedField.getConditionalFormats())
+        {
+            ConditionalFormat actualFormat = intendedField.getConditionalFormats().stream()
+                    .filter(a-> a.toJSON().equals(intendedFormat.toJSON())).findFirst().orElse(null);
+            assertNotNull("conditional formats did not export with full fidelity", actualFormat);
+        }
+        // would like to do validators, but this part of the remoteAPI is incomplete; validators are settable
+        // on FieldDefinition, but not gettable on PropertyDescriptor
```

Comment:
There's no reason we couldn't add a constructor to `FieldDefinition` that takes a `JSONObject` similar to the one in `PropertyDescriptor`.

Reply:
I want this, but am worried about feature-creep in the context of this story.  Perhaps in another story?